app.rb

```diff
@@ -0,0 +1,76 @@
+require 'sinatra/base'
+require 'sinatra/flash'
+
+require_relative 'data_mapper_setup.rb'
+require_relative 'lib/peep'
+require_relative 'lib/user'
+
+class ChitterApp < Sinatra::Base
+
+  setup_databases
+
+  enable :sessions
+  register Sinatra::Flash
+
+  get '/' do
+    'Hello world'
+  end
+
+  post '/users' do
+    string = "Sorry, that email or username has already been used."
+    unless (user = User.make(email: params["email"], password: params["password"], username: params["username"], name: params["name"]))
+      flash[:error] = string
+      redirect '/users/new'
+    else
+      User.sign_in(username: params["username"], password: params["password"])
+      redirect "/users/#{user.id}"
+    end
+  end
+
+  get '/users/new' do
+    erb :sign_up
+  end
+
+  get '/users/sign_in' do
+    erb :sign_in
+  end
+
+  post '/users/sign_in' do
+    string = "Sorry, either your username or password were incorrect"
```

Comment:
Avoid names like 'string'. The name of a variable should reflect the intent - the intent here is error_message or validation_error or something like that. 

That said, do you even need to assign this to a variable in the first place? `flash[:error] = "Sorry, either your username or password were incorrect"` below is fine.